### PR TITLE
Fix move console left/right command

### LINKED ISSUE #7405: View / Panes / Console on Left or Right not working properly (1.4)
### System details

    RStudio Edition : Desktop or Server
    RStudio Version : 1.4.618
    OS Version      : any
    R Version       : any

The same issue was logged and fixed in 1.3 (tracked by #7246). That fix was not sufficient for 1.4 due (presumably) to the work in progress on source columns, so opening this to track for 1.4.

### Steps to reproduce the problem
- startup IDE in a clean default state, with no files open in source pane (for best chance of repro, delete all previous settings including Qt-related settings if using Desktop)
- in View / Panes menu, note checkmark on "Console on Left" menu item
- select View / Panes / Console on Right

### Describe the problem in detail
- an editor document is opened (which is consistent with earlier versions) but the Console remains in lower-left corner
- View / Panes / Console on Right is now checked in the menu despite Console not being on right
- Problem persists even after quit and restart (i.e. Console in lower-left, menu says it's on right)

### Describe the behavior you expected
- Untitled editor document created
- Console panel switches places with Environment panel
- Menu checkbox matches actual visual state including after a restart

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [x] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### PR DESCRIPTION
**Pull request by gtritchie**
Without this, the View / Panes / Console on Left (or Right) commands didn't change anything until you reloaded the UI.

Fixes #7405

(Was also dependent on fix made in v1.3, #7399)

### CODE REVIEW

File: src/gwt/src/org/rstudio/studio/client/workbench/ui/PaneManager.java [hunk 1]
```diff
@@ -403,8 +403,8 @@ else if (numDocs > 0 && window.getState() == WindowState.HIDE)
 
       eventBus.addHandler(UpdateTabPanelsEvent.TYPE, event ->
       {
-         left_.replaceWindows(panes_.get(0), panes_.get(1));
-         right_.replaceWindows(panes_.get(2), panes_.get(3));
+         left_.replaceWindows(panes_.get(0), panes_.get(1), DualWindowLayoutPanel.ReplaceMode.IF_DIFFERENT);
+         right_.replaceWindows(panes_.get(2), panes_.get(3), DualWindowLayoutPanel.ReplaceMode.IF_DIFFERENT);
```
Comment (melissa-barca) [inline src/gwt/src/org/rstudio/studio/client/workbench/ui/PaneManager.java:407]: Looking into it, we don't seem to ever fire `UpdateTabPanelsEvent` so I think we should get rid of this handler and the event.

File: src/gwt/src/org/rstudio/core/client/layout/DualWindowLayoutPanel.java [hunk 2]
```diff
@@ -510,10 +510,16 @@ private void unhookEvents()
...
+      IF_DIFFERENT,
+      ALWAYS
+   }
    public void replaceWindows(LogicalWindow windowA,
-                              LogicalWindow windowB)
+                              LogicalWindow windowB,
+                              ReplaceMode replaceMode)
    {
-      if (windowA_ != windowA && windowB_ != windowB)
+      if (replaceMode == ReplaceMode.ALWAYS || (windowA_ != windowA && windowB_ != windowB))
```
Comment (melissa-barca) [inline src/gwt/src/org/rstudio/core/client/layout/DualWindowLayoutPanel.java:522]: I know you didn't write the original code, but do you know why the comparisons between windows A & B are compared with `&&` as opposed to `||`? It seems like as long as one is different we want to do the replace.

Comment (gtritchie) [reply src/gwt/src/org/rstudio/core/client/layout/DualWindowLayoutPanel.java:522]: You added that comparison in this commit:

11367597c893c239b6c0acde6a4e9d56e49e9905

Comment (melissa-barca) [reply src/gwt/src/org/rstudio/core/client/layout/DualWindowLayoutPanel.java:522]: Wow ok, I think this is a bug and should be changed.

File: src/gwt/src/org/rstudio/studio/client/workbench/ui/PaneManager.java (hunk 1 above)
Comment (gtritchie) [reply src/gwt/src/org/rstudio/studio/client/workbench/ui/PaneManager.java]: Done!

File: src/gwt/src/org/rstudio/core/client/layout/DualWindowLayoutPanel.java (hunk 2 above)
Comment (gtritchie) [reply src/gwt/src/org/rstudio/core/client/layout/DualWindowLayoutPanel.java]: Per discussion I have removed that comparison completely. Thanks!

**Pull request merged**